crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -163,23 +164,6 @@ impl<'w> EntityMut<'w> {
         unsafe { self.get_unchecked_mut::<T>() }
     }
 
-    /// Gets an immutable reference to the component of type `T` associated with
-    /// this entity without ensuring there are no unique borrows active and without
-    /// ensuring that the returned reference will stay valid.
-    ///
-    /// # Safety
-    ///
-    /// - The returned reference must never alias a mutable borrow of this component.
-    /// - The returned reference must not be used after this component is moved which
-    ///   may happen from **any** `insert_component`, `remove_component` or `despawn`
-    ///   operation on this world (non-exhaustive list).
-    #[inline]
-    pub unsafe fn get_unchecked<T: Component>(&self) -> Option<&'w T> {
```

Comment:
This shouldn't be removed in this PR, It's not my favorite API, but we should discuss that separately.

Reply:
Personally I disagree, I see no reason to keep an API that boils down to tacking on an unjustified lifetime transmute onto an otherwise safe method. There are zero use cases for this API that aren't better served by a safe api, and I think it is fitting to remove it in a PR dedicated to increasing soundness in bevy_ecs